### ISSUE TITLE
Cuckoo not actually opening submitted URLs?

### ISSUE THREAD

**Issue by braimee** [cuckoosandbox/cuckoo#2291]
Thanks for creating an issue! But first: did you read our community guidelines?
https://cuckoo.sh/docs/introduction/community.html

##### My issue is:
If I submit just a URL via the Cuckoo interface, I don't think it is actually being analyzed/processed.

##### My Cuckoo version and operating system are:
Ubuntu 16.04, Cuckoo 2.0.5.

##### This can be reproduced by:
Submitting a URL to the Cuckoo Web GUI and monitoring the console log.

##### The log, error, files etc can be found at:
Console log says (and sorry, I only have a snippet of the error):

````
    if not signature.matched and handler(*args, **kwargs):
  File "/home/cuckoo/.cuckoo/signatures/windows/office_rtf.py", line 27, in on_complete
    filetype = self.get_results("target", {})["file"]["type"]
KeyError: 'file'
2018-05-30 15:47:01,161 [cuckoo.core.plugins] ERROR: Failed to run 'on_complete' of the url_file signature
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/cuckoo/core/plugins.py", line 413, in call_signature
    if not signature.matched and handler(*args, **kwargs):
  File "/home/cuckoo/.cuckoo/signatures/windows/url_file.py", 
````

When I look at the report screen once the URL is analyzed, I don't think the URL has actually been visited.  I see nothing in the IP results or DNS lookups indicating that the URL I submitted was actually visited.  Wondering if the error above indicates something else is going on?

FWIW if I submit malware samples the Cuckoo box *does* process them correctly.

**Comment by RicoVZ** [comment]
Hi braimee,

Thanks for posting an issue.

It is possible that the URLs are not being visited. It can be caused by a few reasons.
Obvious question, but I will still ask: do your guest machines have an internet connection?

What browser is installed/used for your URL analyses?

Can you try to re-submit one of the URL analyses, but disable 'Enable Injection' in the web interface.
Like this:
![image](https://user-images.githubusercontent.com/8687431/40750631-3262abd6-6468-11e8-8131-0acb0f91e271.png)

After you have done this, please post if there is a difference.

**About the error**: 
It is unrelated, but still an error. It is caused by a Cuckoo signature that has an error in it. It does not affect the analysis results, except for the fact that this specific signature will not be triggered.

A fix in cuckoosandbox/community has already been made for this and should me merged soon.

As soon as it is merged, you can use `cuckoo community` to update your signatures.

**Comment by braimee** [comment]
Hi @RicoVZ !

Thanks for your quick response.  The default browser on the system is IE, and yes, if I fire up that VM I can verify that IP/DNS connectivity works just fine and I can browse around the Web as expected.

I tried to turn off the injection as you noted, and the console still had an error and the URL did not appear to be actually processed by Cuckoo.  I can see from the report output that lots of *.microsoft.com sites and *.live.com sites were hit during the analysis, but nothing on the actual URL I submitted.

Per the end of your message, it sounds like I should run "cuckoo community" and try again, so I'll do that and report back!

**Comment by braimee** [comment]
OK so here's what I did:

1. Killed Cuckoo process
2. Ran `cuckoo community`
3. Started Cuckoo.
4. Submitted a handful of URLs for analysis with default settings.  At some point, each one coughed up something similar to this:

````
018-06-01 14:55:33,660 [cuckoo.core.plugins] ERROR: Failed to run 'on_complete' of the rtf_unknown_character_set signature
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/cuckoo/core/plugins.py", line 413, in call_signature
    if not signature.matched and handler(*args, **kwargs):
  File "/home/cuckoo/.cuckoo/signatures/windows/office_rtf.py", line 55, in on_complete
    filetype = self.get_results("target", {})["file"]["type"]
KeyError: 'file'
2018-06-01 14:55:33,674 [cuckoo.core.plugins] ERROR: Failed to run 'on_complete' of the rtf_unknown_version signature
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/cuckoo/core/plugins.py", line 413, in call_signature
    if not signature.matched and handler(*args, **kwargs):
  File "/home/cuckoo/.cuckoo/signatures/windows/office_rtf.py", line 27, in on_complete
    filetype = self.get_results("target", {})["file"]["type"]
KeyError: 'file'
2018-06-01 14:55:33,707 [cuckoo.core.plugins] ERROR: Failed to run 'on_complete' of the url_file signature
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/cuckoo/core/plugins.py", line 413, in call_signature
    if not signature.matched and handler(*args, **kwargs):
  File "/home/cuckoo/.cuckoo/signatures/windows/url_file.py", line 21, in on_complete
    if "Internet shortcut" not in self.file.get("type", ""):
AttributeError: 'URLFile' object has no attribute 'file'
````

So then I ran the same URLs with the injection turned off, and the output had similar errors.

Again, I'm not 100% sure that Cuckoo isn't hitting the URLs as I see things like `HTTP Requests (50 out of 3900` (where can I see the full log?) but I've sent Cuckoo a bunch of URLs from malware/phish analysis sites and am pretty sure they should be flagged.

Anything else I can try?  Maybe it would be worth watching the VM fire up "live" and see if IE pops open and actually visits the URL?

Thanks,
Brian

**Comment by doomedraven** [comment]
Yes you should, could you post analysis log from vm and from cuckoo?

**Comment by braimee** [comment]
Hrmmm, interestingly in the Cuckoo GUI, the analysis log is empty, but I attached cuckoo log.
[cuckoolog.txt](https://github.com/cuckoosandbox/cuckoo/files/2065187/cuckoolog.txt)

**Comment by doomedraven** [comment]
im not sure but it looks like it can't speak back to cuckoo host, can you start any sniffer on host and verify what vm talks to port 2042 on host?

**Comment by braimee** [comment]
Sorry could you elaborate a little on that?  Are you wanting to make sure the Cuckoo server is listening on 2042?  It appears to be:

<img width="451" alt="cuckooports" src="https://user-images.githubusercontent.com/17596787/40888743-2d715d28-6721-11e8-991f-5162d77e5e5e.PNG">

**Comment by doomedraven** [comment]
no, what cuckoo server side receives data from client(vm) 

example, you can put more filters see man
`tcpdump port 2042`

**Comment by braimee** [comment]
Ok here's what I saw when running `tcpdump port 2042` during a URL analysis:

````
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on vboxnet0, link-type EN10MB (Ethernet), capture size 262144 bytes
22:42:39.944237 IP 192.168.87.1.2042 > 192.168.87.87.50317: Flags [F.], seq 174459106, ack 2287148871, win 229, length 0
22:42:53.368799 IP 192.168.87.87.50317 > 192.168.87.1.2042: Flags [S], seq 2287164854, win 64240, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
22:42:53.368844 IP 192.168.87.1.2042 > 192.168.87.87.50317: Flags [.], ack 1, win 229, length 0
22:42:53.369277 IP 192.168.87.87.50317 > 192.168.87.1.2042: Flags [R], seq 2287148871, win 0, length 0
22:42:53.589399 IP 192.168.87.87.50318 > 192.168.87.1.2042: Flags [S], seq 1926018100, win 64240, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
22:42:53.589454 IP 192.168.87.1.2042 > 192.168.87.87.50318: Flags [S.], seq 1447374796, ack 1926018101, win 29200, options [mss 1460,nop,nop,sackOK,nop,wscale 7], length 0
22:42:53.589896 IP 192.168.87.87.50318 > 192.168.87.1.2042: Flags [.], ack 1, win 2053, length 0
22:42:53.590929 IP 192.168.87.87.50318 > 192.168.87.1.2042: Flags [P.], seq 1:5, ack 1, win 2053, length 4
22:42:53.590959 IP 192.168.87.1.2042 > 192.168.87.87.50318: Flags [.], ack 5, win 229, length 0
````

**Comment by RicoVZ** [comment]
Thanks for the updates @braimee.

Could you post the following:
- The Internet Explorer version used in the guest machine
- The analysis.log, if it is there (this should help us). Located at: `$CWD/storage/analyses/<taskid>/analyis.log`.

I expect the browser might be crashing upon injection, as it does seem to work when disabling it?

Lastly, can you try to submit a URL like this, and see if there is a difference: `cuckoo submit --url "yoururl"`

**Comment by braimee** [comment]
Thanks for your help @RicoVZ .  Some responses:

* The version of IE on the guest machine is 11.

* I spun up a new task (via the GUI, with injection *enabled* to have a URL analyzed.  It came up as task #50, but when I go to the corresponding folder, the `analysis.log` is there but it's empty.  Not sure why?

In the console, here's the gist of what I saw:

````
[06/Jun/2018 08:13:04] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 744
[06/Jun/2018 08:13:07] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 744
2018-06-06 08:13:07,432 [cuckoo.core.guest] INFO: cuckoo1: end of analysis reached!
2018-06-06 08:13:09,194 [cuckoo.processing.behavior] WARNING: Analysis results folder does not contain any behavior log files.
[06/Jun/2018 08:13:09] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 785
[06/Jun/2018 08:13:12] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 785
2018-06-06 08:13:22,982 [cuckoo.core.plugins] ERROR: Failed to run 'on_complete' of the rtf_unknown_character_set signature
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/cuckoo/core/plugins.py", line 413, in call_signature
    if not signature.matched and handler(*args, **kwargs):
  File "/home/cuckoo/.cuckoo/signatures/windows/office_rtf.py", line 55, in on_complete
    filetype = self.get_results("target", {})["file"]["type"]
KeyError: 'file'
2018-06-06 08:13:22,991 [cuckoo.core.plugins] ERROR: Failed to run 'on_complete' of the rtf_unknown_version signature
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/cuckoo/core/plugins.py", line 413, in call_signature
    if not signature.matched and handler(*args, **kwargs):
  File "/home/cuckoo/.cuckoo/signatures/windows/office_rtf.py", line 27, in on_complete
    filetype = self.get_results("target", {})["file"]["type"]
KeyError: 'file'
2018-06-06 08:13:23,022 [cuckoo.core.plugins] ERROR: Failed to run 'on_complete' of the url_file signature
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/cuckoo/core/plugins.py", line 413, in call_signature
    if not signature.matched and handler(*args, **kwargs):
  File "/home/cuckoo/.cuckoo/signatures/windows/url_file.py", line 21, in on_complete
    if "Internet shortcut" not in self.file.get("type", ""):
AttributeError: 'URLFile' object has no attribute 'file'
[06/Jun/2018 08:13:24] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 785
2018-06-06 08:13:25,304 [cuckoo.core.scheduler] INFO: Task #50: reports generation completed
2018-06-06 08:13:25,317 [cuckoo.core.scheduler] INFO: Task #50: analysis procedure completed
[06/Jun/2018 08:13:27] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 784
````

* I ran the analysis again but with URL injection *disabled* and the log output is similar, and the `analysis.log` is empty.

* I ran a final job with `cuckoo submit --url "yoururl"` and the message I see is:

````
Success: URL "http://domain.com/" added as task with ID #3
````

I might need to either be patient and/or RTFM some more, because I don't see this URL submission as a pending job in the GUI, nor do I see the VM spinning up in the background to conduct an analysis.

**Comment by RicoVZ** [comment]
@braimee
Ah, also still the signature errors. I provided you with an incorrect signature update command earlier. To update existing signatures, please use `cuckoo community --force`. The `--force` flag will make sure existing signature are overwritten with new ones.

I suspect there might be multiple issues here.
First, I would suggest to use a lower version of Internet Explorer. I use IE 8 for URL tasks myself. This one is known to work with the Cuckoo monitor.

After trying this, please add another URL task like so: `cuckoo submit --url "yoururl"`, and start Cuckoo to see if there is any difference.

**Comment by braimee** [comment]
Ah shoot, sorry @RicoVZ I realize now I'm probably using an unsupported config.  I have a Win10 machine, 64-bit.  Not sure if that's the issue.  Anyway, I can't run IE 8 or 9 on it.

I did run the `cuckoo community --force` though, and now when I submit a URL, the VM doesn't appear to open the Web browser still, and errors look about the same.

````
n/2018 17:35:42] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 720
2018-06-07 17:35:44,205 [cuckoo.processing.behavior] WARNING: Analysis results folder does not contain any behavior log files.
[07/Jun/2018 17:35:45] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 760
[07/Jun/2018 17:35:47] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 760
[07/Jun/2018 17:35:50] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 760
[07/Jun/2018 17:35:52] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 760
^C
root@cuckoo22:/tmp# [07/Jun/2018 17:35:55] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 760
[07/Jun/2018 17:35:57] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 760
2018-06-07 17:35:58,812 [cuckoo.core.plugins] ERROR: Failed to run 'on_complete' of the rtf_unknown_character_set signature
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/cuckoo/core/plugins.py", line 413, in call_signature
    if not signature.matched and handler(*args, **kwargs):
  File "/home/cuckoo/.cuckoo/signatures/windows/office_rtf.py", line 55, in on_complete
    filetype = self.get_results("target", {})["file"]["type"]
KeyError: 'file'
2018-06-07 17:35:58,821 [cuckoo.core.plugins] ERROR: Failed to run 'on_complete' of the rtf_unknown_version signature
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/cuckoo/core/plugins.py", line 413, in call_signature
    if not signature.matched and handler(*args, **kwargs):
  File "/home/cuckoo/.cuckoo/signatures/windows/office_rtf.py", line 27, in on_complete
    filetype = self.get_results("target", {})["file"]["type"]
KeyError: 'file'
2018-06-07 17:35:58,852 [cuckoo.core.plugins] ERROR: Failed to run 'on_complete' of the url_file signature
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/cuckoo/core/plugins.py", line 413, in call_signature
    if not signature.matched and handler(*args, **kwargs):
  File "/home/cuckoo/.cuckoo/signatures/windows/url_file.py", line 21, in on_complete
    if "Internet shortcut" not in self.file.get("type", ""):
AttributeError: 'URLFile' object has no attribute 'file'
[07/Jun/2018 17:36:00] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 760
2018-06-07 17:36:01,248 [cuckoo.core.scheduler] INFO: Task #57: reports generation completed
2018-06-07 17:36:01,261 [cuckoo.core.scheduler] INFO: Task #57: analysis procedure completed
[07/Jun/2018 17:36:02] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 759
[07/Jun/2018 17:36:05] "POST /analysis/api/tasks/info/ HTTP/1.1" 200 759

````

Should I shift to a Win 7 VM?  I quick look around the issues list here seems to suggest Win 10 should still work, but I won't waste more time on this if you think Win 7 would help.  Also I saw that a new version of Cuckoo is out.  Should I try the upgrade just for grins?

Thanks!
Brian

**Comment by jbremer** [comment]
We (still) don't officially support 64-bit Windows 10, so downgrading to Windows 7 would make sense :-)
We've updated the broken Signatures, please re-run `cuckoo community -f`.

**Comment by hackdefendr** [comment]
Win10 x64 works fine under Cuckoo 2.0.6.2.  A trick I found somewhere on the net was a registry file that officially removes Edge from Windows 10, thus forcing IE to be the default browser.  This fixes the issue where IE doesn't show up in screenshots, at least for me.  

I can't remember where I found the registry file, but its out there, the file I downloaded was named uninstall_edge.zip.  Run the registry batch file on your VM, then recreate your snapshot. Simply changing the default browser in Win10 doesn't actually do it.
